test/test_video_player.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:video_player_platform_interface/video_player_platform_interface.dart';
+
+void setupMockVideoPlayerPlatform() {
+  VideoPlayerPlatform.instance = MockVideoPlayerPlatform();
+}
+
+class MockVideoPlayerPlatform extends Fake
+    with MockPlatformInterfaceMixin
+    implements VideoPlayerPlatform {
+  static const int _textureId = 0xffffffff;
+
+  static StreamController<VideoEvent> _streamController = StreamController<VideoEvent>();
+  static bool initialized = false;
+  static bool isPlaying = false;
+
+  static void reset() {
+    _streamController.close();
+    _streamController = StreamController<VideoEvent>();
+    initialized = false;
+    isPlaying = false;
+  }
+
+  @override
+  Future<void> init() async {}
+
+  @override
+  Future<void> dispose(int textureId) async {
+    assert(textureId == _textureId);
+    initialized = false;
+  }
+
+  @override
+  Future<int?> create(DataSource dataSource) async  {
+    initialized = true;
+    _streamController.add(VideoEvent(
+      eventType: VideoEventType.initialized,
+      duration: const Duration(seconds: 1),
+      size: const Size(0, 0),
+      rotationCorrection: 0,
+    ));
+    return _textureId;
+  }
+
+  @override
+  Stream<VideoEvent> videoEventsFor(int textureId) {
+    assert(textureId == _textureId);
+    return _streamController.stream;
+  }
+
+  @override
+  Future<void> setLooping(int textureId, bool looping) async {
+    assert(textureId == _textureId);
+  }
+
+  @override
+  Future<void> play(int textureId) async {
+    assert(textureId == _textureId);
+    isPlaying = true;
+    _streamController.add(VideoEvent(
+      eventType: VideoEventType.isPlayingStateUpdate,
+      isPlaying: true,
+    ));
+  }
+
+  @override
+  Future<void> pause(int textureId) async {
+    assert(textureId == _textureId);
+    isPlaying = false;
+    _streamController.add(VideoEvent(
+      eventType: VideoEventType.isPlayingStateUpdate,
+      isPlaying: false,
+    ));
+  }
+
+  @override
+  Future<void> setVolume(int textureId, double volume) async {
+    assert(textureId == _textureId);
+  }
+
+  // Future<void> seekTo(int textureId, Duration position) {
+  //   throw UnimplementedError('seekTo() has not been implemented.');
+  // }
+
+  @override
+  Future<void> setPlaybackSpeed(int textureId, double speed) async {
+  }
+
+  // Future<Duration> getPosition(int textureId) {
+  //   throw UnimplementedError('getPosition() has not been implemented.');
```

Comment:
We can leave out this commented-out code. If/when we write a test where the method does get called, it'll be easy to add a similar stub then; if not, we don't need it.